packages/newspringchurchapp/src/prayer/AddPrayer/AddPrayerForm/AddPrayerForm.js

```diff
@@ -0,0 +1,131 @@
+import React, { memo } from 'react';
+import PropTypes from 'prop-types';
+import { View, KeyboardAvoidingView } from 'react-native';
+import { SafeAreaView } from 'react-navigation';
+import { Formik } from 'formik';
+import {
+  ModalView,
+  Switch,
+  TextInput,
+  Button,
+  styled,
+  FlexedView,
+  PaddedView,
+  H3,
+  H5,
+  Avatar,
+} from '@apollosproject/ui-kit';
+
+const FlexedSafeAreaView = styled({
+  flex: 1,
+})(SafeAreaView);
+
+const ShrinkingView = styled(({ theme }) => ({
+  flex: 1,
+  paddingTop: theme.sizing.baseUnit,
+}))(KeyboardAvoidingView);
+
+const HeaderView = styled({
+  alignItems: 'center',
+})(View);
+
+const TitleView = styled(({ theme }) => ({
+  marginTop: theme.sizing.baseUnit * 0.5,
+  marginBottom: theme.sizing.baseUnit * 2,
+}))(View);
+
+const StyledTextInput = styled({
+  height: '100%',
+})(TextInput);
+
+const StyledButton = styled({
+  borderRadius: 0,
+})(Button);
+
+const BottomView = styled({
+  justifyContent: 'flex-end',
+})(FlexedView);
+
+const SwitchContainer = styled({
+  width: '70%',
+  alignSelf: 'center',
+  flexDirection: 'row',
+  justifyContent: 'center',
+})(PaddedView);
+
+const SwitchLabel = styled(({ theme }) => ({
+  color: theme.colors.primary,
+}))(H5);
+
+const SwitchLabelContainer = styled(({ theme }) => ({
+  justifyContent: 'center',
+  marginLeft: theme.sizing.baseUnit,
+}))(View);
+
+const AddPrayerForm = memo(({ imgSrc, title, btnLabel, ...props }) => (
+  <Formik
+    initialValues={{ prayer: '', anonymous: false }}
+    onSubmit={(values) => console.log(values)}
+  >
+    {({ handleChange, handleBlur, handleSubmit, values }) => (
+      <ModalView {...props}>
+        <FlexedSafeAreaView forceInset={{ top: 'always', bottom: 'never' }}>
+          <ShrinkingView behavior={'padding'}>
+            <HeaderView>
+              {/* TODO: margin will be fixed pending core #667 */}
+              <Avatar
+                source={values.anonymous ? null : imgSrc}
+                size={'medium'}
+              />
+              <TitleView>
+                <H3>{title}</H3>
+              </TitleView>
+            </HeaderView>
+            {/* TODO: add underline={false} prop to TextInput, 
+        pending https://github.com/ApollosProject/apollos-prototype/issues/629 */}
+            <FlexedView>
+              <PaddedView>
+                <StyledTextInput
+                  editable
+                  multiline
+                  placeholder="Start typing your prayer..."
+                  onChangeText={handleChange('prayer')}
+                  onBlur={handleBlur('prayer')}
+                  value={values.prayer}
+                />
+              </PaddedView>
+            </FlexedView>
+            <BottomView>
+              <SwitchContainer>
+                <Switch
+                  value={values.anonymous}
+                  onValueChange={handleChange('anonymous')}
+                />
+                <SwitchLabelContainer>
```

Comment:
Does our `Switch` component not support labels out of the box? or was the built-in label just not styled how you wanted it?

Reply:
it just wasn't styled like we needed. the label was on the wrong side. I could have done a option PR to core but it seems minor.

Reply:
Do we lose any functionality doing it this way? Ie...is the label in the actual component tappable? Also, as the designer on this, I can say I don't think any thought was put into which side of the switch to put the label on...

Reply:
true. we do lose that. I do however think it looks better on the side you put it on. I could move forward with a PR to core to give the option if you think it's worth it?

Reply:
For MVP perspective, I'd probably use the built-in component functionality, and open an issue on core to handle this use case, but I don't think we need to start it quite yet. You're welcome to take a crack at it though

Reply:
oh and also I couldn't change the color

Reply:
Yeah. again, I'd probably use the built-in component for MVP, and we can file an issue for the use case on core and figure out how we want to change the api of the built-in Switch component to handle